### ISSUE TITLE
State Color not working with some entities.

### ISSUE THREAD

**Issue by sheminasalam** [alexarch21/history-explorer-card#51]
I have the following yaml config for state color. 
```
stateColors:
  'off': '#3F5E98'
  'on': '#FFDB15'
  binary_sensor.off: '#3F5E98'
  binary_sensor.on: '#FFDB15'
  person.home: '#3F5E98'
  alarm_control_panel.disarmed
```
With this code, not all binary sensors are adapting the set state colors. Only some binary sensor entities show with the set colors, rest shows with the default colors. why is it so?

**Comment by alexarch21** [comment]
Impossible to say without a lot more information I'm afraid.

**Comment by sheminasalam** [comment]
please let me know what all info you would need

**Comment by sheminasalam** [comment]
full yaml
```
type: custom:history-explorer-card
cardName: historycard-96473670
recordedEntitiesOnly: true
defaultTimeRange: 4h
combineSameUnits: true
header: ' '
stateColors:
  'off': '#3F5E98'
  'on': '#FFDB15'
  binary_sensor.off: '#3F5E98'
  binary_sensor.on: '#FFDB15'
  person.home: '#3F5E98'
  alarm_control_panel.disarmed: '#3F5E98'
  alarm_control_panel.armed_night: '#FFDB15'
  alarm_control_panel.armed_away: red
labelAreaWidth: 150
```
Entity names and integration details
```
binary_sensor.kseb_power_l1  >  mqtt binary sensor
binary_sensor.kseb_power_l2  >  mqtt binary sensor
binary_sensor.kseb_power_l3  >  mqtt binary sensor
binary_sensor.master_switch  >  mqtt binary sensor
binary_sensor.internet_status > ping binary sensor

```
HA screenshots
![image](https://user-images.githubusercontent.com/45196177/181232258-78fc0671-84b0-43eb-a4ac-0efff20a2ce6.png)
![image](https://user-images.githubusercontent.com/45196177/181232951-ff7837b7-c33c-4067-88ec-fac2c409aa33.png)
![image](https://user-images.githubusercontent.com/45196177/181233006-c68c1d95-d583-4b21-bd68-f8b5723d1974.png)
![image](https://user-images.githubusercontent.com/45196177/181233097-114a81db-5784-425c-a633-6bc2ebbd69f8.png)
![image](https://user-images.githubusercontent.com/45196177/181233182-6a5d2589-7aff-499d-b1b9-e69facc6cfb8.png)

No errors in log about this card

**Comment by alexarch21** [comment]
For the screenshots, I meant more one of these sensors in the states entity list where it appears with it's fully scoped domain name. Basically making sure you're actually dealing with the binary_sensor domain here, not entities in the sensor domain that just act as if they're binary. That happens a lot, especially with MQTT.

**Comment by sheminasalam** [comment]
![image](https://user-images.githubusercontent.com/45196177/181236174-e419db4a-8d5e-478c-9fdd-eaf9cb569e16.png)

**Comment by alexarch21** [comment]
Okay, I'll look into it.

**Comment by alexarch21** [comment]
I was able to reproduce this. It will be fixed in the next release.

**Comment by alexarch21** [comment]
Fixed in V1.0.24.
